addons/im_livechat/static/tests/chat_hub_compact.test.js

```diff
@@ -0,0 +1,45 @@
+import { Command, serverState } from "@web/../tests/web_test_helpers";
+import { click, contains } from "@mail/../tests/mail_test_helpers_contains";
+import { defineLivechatModels } from "@im_livechat/../tests/livechat_test_helpers";
+import { setupChatHub, start, startServer } from "@mail/../tests/mail_test_helpers";
+import { withGuest } from "@mail/../tests/mock_server/mail_mock_server";
+import { test } from "@odoo/hoot";
+import { rpc } from "@web/core/network/rpc";
+
+defineLivechatModels();
+test("Do not open chat windows automatically when chat hub is compact", async () => {
+    const pyEnv = await startServer();
+    setupChatHub({ folded: [pyEnv["discuss.channel"].create({ name: "General" })] });
+    const guestId = pyEnv["mail.guest"].create({ name: "Visitor" });
+    const channelId = pyEnv["discuss.channel"].create({
+        channel_member_ids: [
+            Command.create({ partner_id: serverState.partnerId }),
+            Command.create({ guest_id: guestId }),
+        ],
+        channel_type: "livechat",
+        livechat_active: true,
+        livechat_operator_id: serverState.partnerId,
+        create_uid: serverState.publicUserId,
+    });
+    await start();
+    await click("button.fa.fa-ellipsis-h[title='Chat Options']");
+    await click("button.o-mail-ChatHub-option", { text: "Hide all conversations" });
+    await contains(".o-mail-ChatHub-hiddenBtnIcon");
```

Comment:
I would not use `.o-mail-ChatHub-hiddenBtnIcon`, classname is poorly named and is simply used to share style between compact and hidden buttons.

I would use `.o-mail-ChatHub-bubbleBtn.fa-commenting` instead.